llvm/lib/Target/RISCV/RISCVInstrInfoXCheri.td

```diff
@@ -1303,6 +1307,13 @@ multiclass PseudoCheriCmpXchgPat<string Op, Pseudo CmpXchgInst,
             (CmpXchgInst GPCR:$addr, ValTy:$cmp, ValTy:$new, 7)>;
 }
 
+// The explicit address mode atomics are always relaxed
+multiclass PseudoCheriCmpXchgPatExplicit<string Op, Pseudo CmpXchgInst,
+                                 RegisterClass ValTy=GPR> {
+  def : Pat<(!cast<PatFrag>(Op#"_monotonic") GPCR:$addr, ValTy:$cmp, ValTy:$new),
+            (CmpXchgInst GPCR:$addr, ValTy:$cmp, ValTy:$new, 2)>;
```

Comment:
The class for `$addr` should be a parameter, even if you don't want to actually add the definitions and implementations for the address-in-cap-mode versions.

Reply:
Sure, any objections to having it as a default parameter though so I don't have to change the uses?